[Clang][ASTImporter] Implement AST import for CXXParenListInitExpr, SubstNonTypeTemplateParmPackExpr, PseudoObjectExpr

ganenkokb-yandex opened the pull request:
Add new visit functions to ASTImporter for CXXParenListInitExpr, SubstNonTypeTemplateParmPackExpr and PseudoObjectExpr.
On CTU analysis there are lot of "cannot import unsupported AST node" for CXXParenListInitExpr, SubstNonTypeTemplateParmPackExpr and PseudoObjectExpr. Problem occurred after full support of Concepts in importer.

ganenkokb-yandex committed:
Implement VisitSubstNonTypeTemplateParmPackExpr

ganenkokb-yandex committed:
Implement VisitPseudoObjectExpr

ganenkokb-yandex committed:
Implement VisitCXXParenListInitExpr

SweetVishnya commented:
@balazske, can you take a look, please?

shafik reviewed (commented):
We should have tests and probably a release note too.

ganenkokb-yandex committed:
Add tests for VisitSubstNonTypeTemplateParmPackExpr

ganenkokb-yandex committed:
Add test for VisitCXXParenListInitExpr

ganenkokb-yandex committed:
Add test for VisitPseudoObjectExpr

ganenkokb-yandex committed:
Code review

Skip null check in VisitPseudoObjectExpr
Fix bad naming in VisitConceptSpecializationExpr
Simplify test for ImportCXXParenListInitExpr

ganenkokb-yandex commented:
@shafik  Could I have your review please?

shafik commented:
> @shafik Could I have your review please?

I was away at the LLVM dev meeting and WG21 meeting, back to back. If you get the approval of a code owner that should usually be ok if you otherwise addressed all the requests by another reviewer.
